### PR TITLE
Migrate core/field_checkbox.js to goog.module syntax

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test`.

### Resolves

Part of #5026

### Proposed Changes

Converts `core/field_checkbox.js` to `goog.module` with ES6 `const`/`let`.

### CODE REVIEW

File: core/field_checkbox.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ FieldCheckbox.prototype.configure_ = function(config) {
 FieldCheckbox.prototype.initView = function() {
   FieldCheckbox.superClass_.initView.call(this);
 
-  Blockly.utils.dom.addClass(
+  addClass(
```
Comment (cpcallen) [inline core/field_checkbox.js:112]: Question: might `dom.addClass` be more readable here?

File: core/field_checkbox.js [hunk 2]
```diff
@@ -220,6 +220,6 @@ FieldCheckbox.prototype.convertValueToBool_ = function(value) {
   }
 };
 
-Blockly.fieldRegistry.register('field_checkbox', FieldCheckbox);
+register('field_checkbox', FieldCheckbox);
```
Comment (cpcallen) [inline core/field_checkbox.js:224]: I'm pretty sure that `fieldRegistry.register` would be more readable here.

**Review by cpcallen — APPROVED** [review]
Approved, but subject to your considering the readability points below.  I think you may have been overly aggressive with the import destructuring.

File: core/field_checkbox.js (hunk 1 above)
Comment (gonfunko) [reply core/field_checkbox.js]: Done - I got some feedback in the opposite direction on a review yesterday, but I do sort of prefer the additional context.

File: core/field_checkbox.js (hunk 2 above)
Comment (gonfunko) [reply core/field_checkbox.js]: Done

**Pull request merged**